xbmc/filesystem/CurlFile.cpp

```diff
@@ -1024,7 +1024,7 @@ bool CCurlFile::Open(const CURL& url)
   // since we can't know the stream size up front if we're gzipped/deflated
   // flag the stream with an unknown file size rather than the compressed
   // file size.
-  if (!m_acceptencoding.empty())
+  if (!m_state->m_httpheader.GetValue("Content-Encoding").empty() && m_state->m_httpheader.GetValue("Content-Encoding") != "identity")
```

Comment:
```suggestion
  if (!m_state->m_httpheader.GetValue("Content-Encoding").empty() && !StringUtils::EqualsNoCase(m_state->m_httpheader.GetValue("Content-Encoding"), "identity"))
```

"All content-coding values are case-insensitive"

Reply:
Thank you for pointing that out - I've updated the commit and force pushed the change.